client/components/domains/domain-product-price/index.jsx

```diff
@@ -93,8 +102,8 @@ const DomainProductPrice = React.createClass( {
 } );
 
 export default connect(
-	state => ( { domainsWithPlansOnly: getCurrentUser( state ) ?
-		currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY ) :
-		true
+	state => ( { domainsWithPlansOnly: getCurrentUser( state )
+		? currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY )
+		: true
```

Comment:
(Obviously, if it still doesn't read well, I'm not gonna defend the binary expr 😄, but I wanted to share my interpretation.)